File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeTwoLists(ListNode* l1, ListNode* l2) {
       if(!l1) return l2;
       if(!l2) return l1;
        
        ListNode* head=l1;
        if(l2->val < l1->val){
            head=l2;
            l2=l2->next;            
        }
        else{
            l1=l1->next;
        }
        ListNode* curr=head;
        
        while(l1 && l2){
        if(l2->val > l1->val){
            curr->next=l1;
            l1=l1->next;
        }
        else{
            curr->next =l2;
            l2=l2->next;
        }
            curr=curr->next;
        }
        if(!l1) curr->next=l2;
        else curr->next=l1;
        
        return head;
    }
};

File: 0021-merge-two-sorted-lists/0021-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        temp = ListNode(-1)
        dummy = temp
        
        while list1 and list2:
            if list1.val <= list2.val:
                newNode = ListNode(list1.val)
                while dummy.next:
                    dummy = dummy.next
                dummy.next = newNode
                list1 = list1.next
            else:
                newNode = ListNode(list2.val)
                while dummy.next:
                    dummy = dummy.next
                dummy.next = newNode 
                list2 = list2.next
                
        while list1:
            newNode = ListNode(list1.val)            
            while dummy.next:
                dummy = dummy.next                    
            dummy.next = newNode
            list1 = list1.next
            
        while list2:
            newNode = ListNode(list2.val)
            while dummy.next:
                dummy = dummy.next
            dummy.next = newNode
            list2 = list2.next
            
       
        return temp.next